Tests/SentryTests/SentryClientTests.swift

```diff
@@ -3,38 +3,186 @@
 import XCTest
 
 class SentryClientTest: XCTestCase {
-    
-    private var client: Client!
-    private var transport: TestTransport!
-    
+
+    private class Fixture {
+        let transport = TestTransport()
+
+        func getSut(configureOptions: (Options) -> Void = { _ in }) -> Client {
+            var client: Client!
+            do {
+                let options = try Options(dict: [
+                    "dsn": TestConstants.dsnAsString
+                ])
+                configureOptions(options)
+
+                client = Client(options: options, andTransport: transport, andFileManager: try SentryFileManager(dsn: TestConstants.dsn))
+            } catch {
+                XCTFail("Options could not be created")
+            }
+
+            return client
+        }
+
+        func getSutWithDisabledSdk() -> Client {
+            getSut(configureOptions: { options in
+                options.enabled = false
+            })
+        }
+
+        func assertEventNotSent(eventId: String?) {
+            XCTAssertNil(transport.lastSentEvent)
+            XCTAssertNil(eventId)
+        }
+
+        func assertLastSentEvent(assert: (Event) -> Void) {
+            if let lastSentEvent = transport.lastSentEvent {
+                assert(lastSentEvent)
+            } else {
+                XCTFail("LastSentEvent must not be nil")
+            }
+        }
+        
+        func assertLastSentEnvelope(assert: (SentryEnvelope) -> Void) {
+            if let lastSentEnvelope = transport.lastSentEnvelope {
+                assert(lastSentEnvelope)
+            } else {
+                XCTFail("LastSentEnvelope must not be nil")
+            }
+        }
+    }
+
+    private let message = "message"
+    private var fixture: Fixture!
+
     override func setUp() {
         super.setUp()
-        
-        transport = TestTransport()
-        
-        do {
-            let options = try Options(dict: [
-                "attachStacktrace": true,
-                "dsn": TestConstants.dsnAsString
-            ])
-    
-            client = Client(options: options, andTransport: transport, andFileManager: try SentryFileManager(dsn: TestConstants.dsn))
-        } catch {
-            XCTFail("Options could not be created")
+        fixture = Fixture()
+    }
+
+    func testCaptureMessage() {
+        let eventId = fixture.getSut(configureOptions: { options in
+            options.attachStacktrace = true
+        }).capture(message: message, scope: nil)
+
+        fixture.assertLastSentEvent { actual in
+            XCTAssertNotNil(eventId)
```

Comment:
Could this assert be done outside of the callback? It doesn't use `actual`

Reply:
Good catch thanks 👍